### ISSUE TITLE
get ipfs to work in javascript

### ISSUE THREAD

**Issue by kingflurkel** [kingflurkel/sponnietube#1]
Uploading and streaming. 
Uploading: https://github.com/ipfspics/server/wiki/How-to-install-ipfs.pics-on-your-own-machine

**Comment by kingflurkel** [comment]
im afraid we'll have to wait on progress on js-ipfs. 
@diasdavid is working on that, I saw a new branch with "add" functionality.

**Comment by daviddias** [comment]
Hey @kingflurkel! You are looking for adding files from a browser tab and make them available through a browser node? 

jsipfs add is on the works (working), right now I'm working to get go-ipfs and js-ipfs nodes to talk with each other through libp2p swarm, then it will be bitswap and we will be able to exchange blocks between browser and go-ipfs nodes :):) so close!!

**Comment by kingflurkel** [comment]
@diasdavid what can we do to help? @sponnet

**Comment by sponnet** [comment]
Yeah, I can look into issues if they can help in the proceedings here..

**Comment by daviddias** [comment]
Awesome! Please read my last 'Captain.log' entry, it has some directions on where your help can be most valuable right now :)

-> https://github.com/ipfs/js-ipfs/issues/30#issuecomment-187950929

I'm available to clarify any details :) :)
